gradle/testing.gradle

```diff
@@ -122,7 +122,7 @@ task testJar(type: Jar, dependsOn: testClasses) {
 
 configurations {
     tests {
-        extendsFrom testRuntime
+        extendsFrom testRuntimeOnly
     }
```

Comment:
This could be replaced by `testFixtures`, with some files moving there from `test`, but I aimed to keep changes to a minimum and limited to Gradle files.